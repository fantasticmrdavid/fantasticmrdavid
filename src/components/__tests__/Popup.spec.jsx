import React from 'react';
import { mount } from 'enzyme';
import Ui from 'components/Popup/ui';
import {
  Close,
  Landscape,
  Lightbox,
  Portrait,
} from 'components/Popup/styles';

const props = {
  active: true,
  close: jest.fn(),
  loading: false,
  orientation: 'landscape',
  title: 'Popup Test',
  url: '/about',
};

describe('Component -> Popup', () => {
  it('should render the loading Popup UI without throwing an error', () => {
    expect(mount(<Ui {...props} loading />)).toMatchSnapshot();
  });

  it('should render the closed Popup UI without throwing an error', () => {
    expect(mount(<Ui {...props} active={false} />)).toMatchSnapshot();
  });

  it('should render the open Popup UI without throwing an error', () => {
    expect(mount(<Ui {...props} />)).toMatchSnapshot();
  });

  it('should call close() when Lightbox is clicked', () => {
    const component = mount(<Ui {...props} />);
    component.find(Lightbox).simulate('click');
    expect(props.close).toHaveBeenCalled();
  });

  it('should call close() when Close element is clicked', () => {
    const component = mount(<Ui {...props} />);
    component.find(Close).simulate('click');
    expect(props.close).toHaveBeenCalled();
  });

  it('should render a Landscape element if a URL is provided and orientation is landscape', () => {
    const component = mount(<Ui {...props} />);
    expect(component.find(Landscape)).toHaveLength(1);
  });

  it('should NOT render a Portrait element if the orientation is landscape', () => {
    const component = mount(<Ui {...props} />);
    expect(component.find(Portrait)).toHaveLength(0);
  });

  it('should render a Portrait element if a URL is provided and orientation is portrait', () => {
    const component = mount(<Ui {...props} orientation="portrait" />);
    expect(component.find(Portrait)).toHaveLength(1);
  });

  it('should NOT render a Landscape element if the orientation is landscape', () => {
    const component = mount(<Ui {...props} orientation="portrait" />);
    expect(component.find(Landscape)).toHaveLength(0);
  });
});
