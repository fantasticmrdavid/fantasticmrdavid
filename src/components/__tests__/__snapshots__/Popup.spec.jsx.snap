// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Popup should render the closed Popup UI without throwing an error 1`] = `
<Memo()
  active={false}
  close={[MockFunction]}
  loading={false}
  orientation="landscape"
  title="Popup Test"
  url="/about"
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-hsnn8o-0",
            "isStatic": false,
            "lastClassName": "jYwInk",
            "rules": Array [
              "position:fixed;width:100%;height:100%;z-index:",
              [Function],
              ";display:flex;justify-content:center;align-items:center;transition:0.5s;opacity:",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-hsnn8o-0",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="styles__Container-hsnn8o-0 jYwInk"
      >
        <styles__Lightbox
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Lightbox-hsnn8o-1",
                  "isStatic": false,
                  "lastClassName": "ksCHkr",
                  "rules": Array [
                    "position:absolute;top:0px;left:0px;width:100%;height:100%;background:",
                    "rgba(0,0,0,0.7)",
                    ";",
                  ],
                },
                "displayName": "styles__Lightbox",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Lightbox-hsnn8o-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <div
              className="styles__Lightbox-hsnn8o-1 ksCHkr"
              onClick={[Function]}
            />
          </StyledComponent>
        </styles__Lightbox>
        <styles__Dialog>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Dialog-hsnn8o-2",
                  "isStatic": false,
                  "lastClassName": "gyNTYq",
                  "rules": Array [
                    "position:relative;border:",
                    "1px solid #333",
                    ";box-shadow:",
                    "7px 7px 5px rgba(0, 0, 0, 0.2)",
                    ";max-width:90%;max-height:90%;transition:0.5s;",
                  ],
                },
                "displayName": "styles__Dialog",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Dialog-hsnn8o-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="styles__Dialog-hsnn8o-2 gyNTYq"
            >
              <styles__Close
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Close-hsnn8o-5",
                        "isStatic": false,
                        "lastClassName": "esqoOc",
                        "rules": Array [
                          "position:absolute;top:0px;right:0px;padding:0.5em;color:white;background-color:",
                          "rgba(0,0,0,0.2)",
                          ";font-size:1.4rem;cursor:pointer;transition:0.3s;z-index:2;",
                          "@media (min-width:",
                          "35.5",
                          "em){",
                          "
    right: -1em;
    padding: 0;
  ",
                          "}",
                          ";",
                          "@media (hover:hover){",
                          "
    &:hover {
      color: tomato;
    }
  ",
                          "}",
                          ";",
                        ],
                      },
                      "displayName": "styles__Close",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Close-hsnn8o-5",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <div
                    className="styles__Close-hsnn8o-5 esqoOc"
                    onClick={[Function]}
                  >
                    X
                  </div>
                </StyledComponent>
              </styles__Close>
              <styles__Landscape
                alt="Popup Test"
                src="/about"
              >
                <StyledComponent
                  alt="Popup Test"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Landscape-hsnn8o-3",
                        "isStatic": false,
                        "lastClassName": "linvvc",
                        "rules": Array [
                          "width:1000px;max-width:90vw;height:auto;",
                          "@media (min-width:",
                          "100",
                          "em){",
                          "
    width: 95vw;
    max-width: 1600px;
  ",
                          "}",
                        ],
                      },
                      "displayName": "styles__Landscape",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Landscape-hsnn8o-3",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="/about"
                >
                  <img
                    alt="Popup Test"
                    className="styles__Landscape-hsnn8o-3 linvvc"
                    src="/about"
                  />
                </StyledComponent>
              </styles__Landscape>
            </div>
          </StyledComponent>
        </styles__Dialog>
      </section>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;

exports[`Component -> Popup should render the loading Popup UI without throwing an error 1`] = `
<Memo()
  active={true}
  close={[MockFunction]}
  loading={true}
  orientation="landscape"
  title="Popup Test"
  url="/about"
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-hsnn8o-0",
            "isStatic": false,
            "lastClassName": "jYwInk",
            "rules": Array [
              "position:fixed;width:100%;height:100%;z-index:",
              [Function],
              ";display:flex;justify-content:center;align-items:center;transition:0.5s;opacity:",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-hsnn8o-0",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="styles__Container-hsnn8o-0 jYwInk"
      >
        <styles__Lightbox
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Lightbox-hsnn8o-1",
                  "isStatic": false,
                  "lastClassName": "ksCHkr",
                  "rules": Array [
                    "position:absolute;top:0px;left:0px;width:100%;height:100%;background:",
                    "rgba(0,0,0,0.7)",
                    ";",
                  ],
                },
                "displayName": "styles__Lightbox",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Lightbox-hsnn8o-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <div
              className="styles__Lightbox-hsnn8o-1 ksCHkr"
              onClick={[Function]}
            />
          </StyledComponent>
        </styles__Lightbox>
        <styles__Dialog>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Dialog-hsnn8o-2",
                  "isStatic": false,
                  "lastClassName": "gyNTYq",
                  "rules": Array [
                    "position:relative;border:",
                    "1px solid #333",
                    ";box-shadow:",
                    "7px 7px 5px rgba(0, 0, 0, 0.2)",
                    ";max-width:90%;max-height:90%;transition:0.5s;",
                  ],
                },
                "displayName": "styles__Dialog",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Dialog-hsnn8o-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="styles__Dialog-hsnn8o-2 gyNTYq"
            >
              <styles__Close
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Close-hsnn8o-5",
                        "isStatic": false,
                        "lastClassName": "esqoOc",
                        "rules": Array [
                          "position:absolute;top:0px;right:0px;padding:0.5em;color:white;background-color:",
                          "rgba(0,0,0,0.2)",
                          ";font-size:1.4rem;cursor:pointer;transition:0.3s;z-index:2;",
                          "@media (min-width:",
                          "35.5",
                          "em){",
                          "
    right: -1em;
    padding: 0;
  ",
                          "}",
                          ";",
                          "@media (hover:hover){",
                          "
    &:hover {
      color: tomato;
    }
  ",
                          "}",
                          ";",
                        ],
                      },
                      "displayName": "styles__Close",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Close-hsnn8o-5",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <div
                    className="styles__Close-hsnn8o-5 esqoOc"
                    onClick={[Function]}
                  >
                    X
                  </div>
                </StyledComponent>
              </styles__Close>
              <styles__Landscape
                alt="Popup Test"
                src="/about"
              >
                <StyledComponent
                  alt="Popup Test"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Landscape-hsnn8o-3",
                        "isStatic": false,
                        "lastClassName": "linvvc",
                        "rules": Array [
                          "width:1000px;max-width:90vw;height:auto;",
                          "@media (min-width:",
                          "100",
                          "em){",
                          "
    width: 95vw;
    max-width: 1600px;
  ",
                          "}",
                        ],
                      },
                      "displayName": "styles__Landscape",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Landscape-hsnn8o-3",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="/about"
                >
                  <img
                    alt="Popup Test"
                    className="styles__Landscape-hsnn8o-3 linvvc"
                    src="/about"
                  />
                </StyledComponent>
              </styles__Landscape>
            </div>
          </StyledComponent>
        </styles__Dialog>
      </section>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;

exports[`Component -> Popup should render the open Popup UI without throwing an error 1`] = `
<Memo()
  active={true}
  close={[MockFunction]}
  loading={false}
  orientation="landscape"
  title="Popup Test"
  url="/about"
>
  <styles__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Container-hsnn8o-0",
            "isStatic": false,
            "lastClassName": "jYwInk",
            "rules": Array [
              "position:fixed;width:100%;height:100%;z-index:",
              [Function],
              ";display:flex;justify-content:center;align-items:center;transition:0.5s;opacity:",
              [Function],
              ";",
            ],
          },
          "displayName": "styles__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Container-hsnn8o-0",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="styles__Container-hsnn8o-0 jYwInk"
      >
        <styles__Lightbox
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Lightbox-hsnn8o-1",
                  "isStatic": false,
                  "lastClassName": "ksCHkr",
                  "rules": Array [
                    "position:absolute;top:0px;left:0px;width:100%;height:100%;background:",
                    "rgba(0,0,0,0.7)",
                    ";",
                  ],
                },
                "displayName": "styles__Lightbox",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Lightbox-hsnn8o-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <div
              className="styles__Lightbox-hsnn8o-1 ksCHkr"
              onClick={[Function]}
            />
          </StyledComponent>
        </styles__Lightbox>
        <styles__Dialog>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__Dialog-hsnn8o-2",
                  "isStatic": false,
                  "lastClassName": "gyNTYq",
                  "rules": Array [
                    "position:relative;border:",
                    "1px solid #333",
                    ";box-shadow:",
                    "7px 7px 5px rgba(0, 0, 0, 0.2)",
                    ";max-width:90%;max-height:90%;transition:0.5s;",
                  ],
                },
                "displayName": "styles__Dialog",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__Dialog-hsnn8o-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="styles__Dialog-hsnn8o-2 gyNTYq"
            >
              <styles__Close
                onClick={[Function]}
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Close-hsnn8o-5",
                        "isStatic": false,
                        "lastClassName": "esqoOc",
                        "rules": Array [
                          "position:absolute;top:0px;right:0px;padding:0.5em;color:white;background-color:",
                          "rgba(0,0,0,0.2)",
                          ";font-size:1.4rem;cursor:pointer;transition:0.3s;z-index:2;",
                          "@media (min-width:",
                          "35.5",
                          "em){",
                          "
    right: -1em;
    padding: 0;
  ",
                          "}",
                          ";",
                          "@media (hover:hover){",
                          "
    &:hover {
      color: tomato;
    }
  ",
                          "}",
                          ";",
                        ],
                      },
                      "displayName": "styles__Close",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Close-hsnn8o-5",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <div
                    className="styles__Close-hsnn8o-5 esqoOc"
                    onClick={[Function]}
                  >
                    X
                  </div>
                </StyledComponent>
              </styles__Close>
              <styles__Landscape
                alt="Popup Test"
                src="/about"
              >
                <StyledComponent
                  alt="Popup Test"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__Landscape-hsnn8o-3",
                        "isStatic": false,
                        "lastClassName": "linvvc",
                        "rules": Array [
                          "width:1000px;max-width:90vw;height:auto;",
                          "@media (min-width:",
                          "100",
                          "em){",
                          "
    width: 95vw;
    max-width: 1600px;
  ",
                          "}",
                        ],
                      },
                      "displayName": "styles__Landscape",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__Landscape-hsnn8o-3",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="/about"
                >
                  <img
                    alt="Popup Test"
                    className="styles__Landscape-hsnn8o-3 linvvc"
                    src="/about"
                  />
                </StyledComponent>
              </styles__Landscape>
            </div>
          </StyledComponent>
        </styles__Dialog>
      </section>
    </StyledComponent>
  </styles__Container>
</Memo()>
`;
